redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisConnectionPoolCache.java

```diff
@@ -0,0 +1,406 @@
+                }
+            } finally {
+                asyncPool.release(connection);
+            }
+        });
+    }
+
+    private List<byte[]> allKeys(RedisKeyCommands<byte[], byte[]> commands, byte[] pattern) {
+        return commands.keys(pattern);
+    }
+
+    @Override
+    public AsyncCache<AsyncPool<StatefulConnection<byte[], byte[]>>> async() {
+        return asyncCache;
+    }
+
+    /**
+     * Get the value based on the parameters.
+     *
+     * @param requiredType  requiredType
+     * @param serializedKey serializedKey
+     * @param <T>           type of the argument
+     * @return value
+     */
+    @Override
+    protected <T> Optional<T> getValue(Argument<T> requiredType, byte[] serializedKey) {
+        try {
+            return asyncPool.acquire().thenCompose(connection -> {
+                try {
+                    RedisStringCommands<byte[], byte[]> stringCommands = getRedisStringCommands(connection);
+                    RedisKeyCommands<byte[], byte[]> keyCommands = getRedisKeyCommands(connection);
+                    return CompletableFuture.completedFuture(getValue(requiredType, serializedKey, stringCommands, keyCommands));
+                } finally {
+                    asyncPool.release(connection);
+                }
+            }).get();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return Optional.empty();
+        }
+    }
+
+    private <T> Optional<T> getValue(
+            Argument<T> requiredType,
+            byte[] serializedKey,
+            RedisStringCommands<byte[], byte[]> stringCommands,
+            RedisKeyCommands<byte[], byte[]> keyCommands
+    ) {
+        byte[] data = stringCommands.get(serializedKey);
+        if (expireAfterAccess != null) {
+            keyCommands.pexpire(serializedKey, expireAfterAccess);
+        }
+        if (data != null) {
+            return valueSerializer.deserialize(data, requiredType);
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Place the value in the cache.
+     *
+     * @param serializedKey serializedKey
+     * @param value         value
+     * @param <T>           type of the value
+     */
+    @Override
+    protected <T> void putValue(byte[] serializedKey, T value) {
+        Optional<byte[]> serialized = valueSerializer.serialize(value);
+
+        try {
+            asyncPool.acquire().thenAccept(connection -> {
+                try {
+                    RedisStringCommands<byte[], byte[]> stringCommands = getRedisStringCommands(connection);
+                    RedisKeyCommands<byte[], byte[]> keyCommands = getRedisKeyCommands(connection);
+                    putValue(serializedKey, serialized, stringCommands, keyCommands);
+                } finally {
+                    asyncPool.release(connection);
+                }
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Needs logging / proper exception handling. It should probably be rethrown as an actual exception instead of swallowed